packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.base.tsx

```diff
@@ -216,15 +216,14 @@ export class CommandBarBase extends BaseComponent<ICommandBarProps, {}> implemen
       overflowItems = [movedItem, ...overflowItems];
       primaryItems = shiftOnReduce ? primaryItems.slice(1) : primaryItems.slice(0, -1);
 
-      data.primaryItems = primaryItems;
-      data.overflowItems = overflowItems;
-      cacheKey = this._computeCacheKey(data);
+      const newData = { ...data, primaryItems, overflowItems };
+      cacheKey = this._computeCacheKey(newData);
 
       if (onDataReduced) {
         onDataReduced(movedItem);
       }
 
-      return { ...data, cacheKey };
+      return { ...newData, cacheKey };
```

Comment:
Just wondering if you really need to spread twice. If cacheKey was part of line 219 (so you swap line 219 and 220), and return that, wouldn't it work?

Reply:
I'm actually just reverting the logic to what it was before (as much as possible so the owners of this code would be able to recognize it :))